src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs

```diff
@@ -558,7 +557,7 @@ internal async Task PublishDiagnosticsAsync(IDiagnosticService diagnosticService
                 {
                     // There were diagnostics from other documents, but none from the current document.
                     // If we're tracking the current document, we can stop.
-                    _publishedFileToDiagnostics.GetOrDefault(fileUri)?.Remove(document.Id);
+                    ((IReadOnlyDictionary<Uri, Dictionary<DocumentId, ImmutableArray<LanguageServer.Protocol.Diagnostic>>>)_publishedFileToDiagnostics).GetValueOrDefault(fileUri)?.Remove(document.Id);
```

Comment:
Hmm -- I think we have another overload that takes a defaultValue parameter that might make it easier to deal with than this.